ttnn/cpp/ttnn/operations/data_movement/slice/slice.cpp

```diff
@@ -8,133 +8,239 @@
     const ttnn::Tensor& input_tensor,
-    tt::tt_metal::Array4D output_tensor_start,
-    tt::tt_metal::Array4D output_tensor_end,
-    const std::optional<tt::tt_metal::Array4D> step,
-    const std::optional<MemoryConfig>& memory_config_arg) {
-    return invoke(DefaultQueueId, input_tensor, output_tensor_start, output_tensor_end, step, memory_config_arg);
-}
+    const std::vector<int> &begins,
+    const std::vector<int> &ends,
+    const std::vector<int> &step,
+    const std::optional<MemoryConfig>& memory_config_arg);
 
-ttnn::Tensor SliceOperation::invoke(
+template ttnn::Tensor SliceOperation::invoke<int>(
     const ttnn::Tensor& input_tensor,
-    tt::tt_metal::Array4D output_tensor_start,
-    tt::tt_metal::Array4D output_tensor_end,
-    const std::optional<tt::tt_metal::Array4D> step) {
-    return invoke(DefaultQueueId, input_tensor, output_tensor_start, output_tensor_end, step, std::nullopt);
-}
+    const std::vector<int> &begins,
+    const std::vector<int> &ends,
+    const std::vector<int> &step,
+    const std::optional<MemoryConfig>& memory_config_arg);
+
+
+template ttnn::Tensor SliceOperation::invoke<uint32_t>(
+    uint8_t queue_id,
+    const ttnn::Tensor& input_tensor,
+    const std::vector<uint32_t> &begins,
+    const std::vector<uint32_t> &ends,
+    const std::vector<uint32_t> &step,
+    const std::optional<MemoryConfig>& memory_config_arg);
+
+template ttnn::Tensor SliceOperation::invoke<uint32_t>(
+    const ttnn::Tensor& input_tensor,
+    const std::vector<uint32_t> &begins,
+    const std::vector<uint32_t> &ends,
+    const std::vector<uint32_t> &step,
+    const std::optional<MemoryConfig>& memory_config_arg);
+
+template ttnn::Tensor SliceOperation::invoke<uint32_t, 4>(
+    uint8_t queue_id,
+    const ttnn::Tensor& input_tensor,
+    const std::array<uint32_t, 4> &output_tensor_start,
+    const std::array<uint32_t, 4> &output_tensor_end,
+    const std::array<uint32_t, 4> &step,
+    const std::optional<MemoryConfig>& memory_config_arg);
+
+template ttnn::Tensor SliceOperation::invoke<uint32_t, 4>(
+    const ttnn::Tensor& input_tensor,
+    const std::array<uint32_t, 4> &output_tensor_start,
+    const std::array<uint32_t, 4> &output_tensor_end,
+    const std::array<uint32_t, 4> &step,
+    const std::optional<MemoryConfig>& memory_config_arg);
+
+template ttnn::Tensor SliceOperation::invoke<uint32_t, 1>(
+    uint8_t queue_id,
+    const ttnn::Tensor& input_tensor,
+    const std::array<uint32_t, 1> &output_tensor_start,
+    const std::array<uint32_t, 1> &output_tensor_end,
+    const std::array<uint32_t, 1> &step,
+    const std::optional<MemoryConfig>& memory_config_arg);
+
+template ttnn::Tensor SliceOperation::invoke<uint32_t, 1>(
+    const ttnn::Tensor& input_tensor,
+    const std::array<uint32_t, 1> &output_tensor_start,
+    const std::array<uint32_t, 1> &output_tensor_end,
+    const std::array<uint32_t, 1> &step,
+    const std::optional<MemoryConfig>& memory_config_arg);
+
```

Comment:
this combinatorial explosion is such a mess in infra. need to fix this :|

Reply:
yeah lol, I wish C++ had a way to combinatorially instantiate